eng/Net9Workarounds.targets

```diff
@@ -0,0 +1,137 @@
+      Include="@(KnownILCompilerPack->WithMetadataValue('TargetFramework', 'net8.0'))"
+      TargetFramework="$(LatestTargetFramework)"
+      Condition="'@(KnownILCompilerPack->Count())' != '0' AND
+        !(@(KnownILCompilerPack->AnyHaveMetadataValue('TargetFramework', '$(LatestTargetFramework)')))" />
+    <KnownWebAssemblySdkPack
+      Include="@(KnownWebAssemblySdkPack->WithMetadataValue('TargetFramework', 'net8.0'))"
+      TargetFramework="$(LatestTargetFramework)"
+      Condition="'@(KnownWebAssemblySdkPack->Count())' != '0' AND
+        !(@(KnownWebAssemblySdkPack->AnyHaveMetadataValue('TargetFramework', '$(LatestTargetFramework)')))" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <!--
+      Reference base shared framework at incoming dependency flow version, not bundled sdk version.
+      Apply this to all projects that target the default tfm (e.g. net8.0) or a rid-based variant of it (e.g. net8.0-windows)
+    -->
+    <FrameworkReference
+      Update="Microsoft.NETCore.App"
+      Condition=" (('$(ProjectTargetFrameworkIdentifier)' == '$(LatestTargetFramework)') AND '$(LatestTargetFramework)' == '$(ProjectTargetFrameworkVersion)') AND '$(TargetLatestDotNetRuntime)' != 'false' "
+      RuntimeFrameworkVersion="$(MicrosoftNETCoreAppRuntimeVersion)"
+      TargetingPackVersion="$(MicrosoftNETCoreAppRefVersion)" />
+
+    <PackageReference Include="Microsoft.DotNet.ILCompiler"
+      Condition=" '$(PublishAot)' == 'true' "
+      Version="$(MicrosoftNETCoreAppRuntimeVersion)" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <!-- Use the same NETCore shared framework as repo built against except when building product code in servicing. -->
+    <KnownFrameworkReference Update="Microsoft.NETCore.App">
+      <LatestRuntimeFrameworkVersion
+          Condition=" '%(TargetFramework)' == '$(LatestTargetFramework)' ">$(MicrosoftNETCoreAppRuntimeVersion)</LatestRuntimeFrameworkVersion>
+      <TargetingPackVersion
+          Condition=" '%(TargetFramework)' == '$(LatestTargetFramework)' ">$(MicrosoftNETCoreAppRefVersion)</TargetingPackVersion>
+      <!--
+        Change the default shared framework version only when _not_ servicing. Avoid bumping version used in most
+        projects. When servicing, projects (Microsoft.AspNetCore.App.Runtime in particular) can use
+        $(TargetLatestRuntimePatch) to explicitly control whether assemblies build against default (false) or
+        latest (true). When that property is not set, SDK uses default metadata in most cases but published apps
+        e.g. tool projects (again, property not set) use latest.
+        On the other hand, $(TargetLatestDotNetRuntime) is specific to this repo and controls only the update below.
+      -->
+      <DefaultRuntimeFrameworkVersion Condition=" '$(IsServicingBuild)' != 'true' AND
+          '%(TargetFramework)' == '$(LatestTargetFramework)' AND
+          '$(TargetLatestDotNetRuntime)' != 'false' ">$(MicrosoftNETCoreAppRuntimeVersion)</DefaultRuntimeFrameworkVersion>
```

Comment:
Remove the `TargetLatestDotNetRuntime` part from this, that's aspnetcore-specific